RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -1,182 +1,75 @@
-/// read and write access to the wallets profile. Every write
-/// of the profile is protected agaist data races thanks to
-/// the actor model and every write persists it in keychain
-/// and if user has enabled: the keychain item syncs to iCloud
-/// as well.
-///
-/// This actor is **not** meant to be used directly by the
-/// apps reducers, but rather always indirectly, via the live
-/// implementations of a set of clients (dependencies), e.g.:
-/// * AccountsClient
-/// * PersonasClient
-/// * AuthorizedDappsClient
-/// * AppPreferencesClient
-///
-/// These live implementaions will all references the one and only
-/// ProfileStore singleton instance `shared`.
-///
-/// Internally, the ProfileStore is a state machine which can be either
-/// in state`ephemeral` or `persisted`. The former state
-/// is used if no ProfileSnapshot was found in `secureStorageClient`,
-/// and will trigger the user to perform onboarding in the wallet. If
-/// a ProfileSnapshot was found instead the state `persisted` is used.
-///
-/// The public interface of the ProfileStore is:
-///
-///     static func shared() async -> ProfileStore
-///     func getLoadProfileOutcome() async -> LoadProfileOutcome
-///     var profile: Profile (async)
-///     func values() -> AnyAsyncSequence<Profile>
-///     func commitEphemeral() async throws
-///     func deleteProfile() async throws
-///     func update(profile: Profile) async throws
-///     func importProfileSnapshot(:ProfileSnapshot) async throws
-///
-///
 public final actor ProfileStore {
 	@Dependency(\.assertionFailure) var assertionFailure
 	@Dependency(\.secureStorageClient) var secureStorageClient
 	@Dependency(\.userDefaultsClient) var userDefaultsClient
 
-	private static let managedAtomicLazyRef = ManagedAtomicLazyReference<ProfileStore>()
-	public static var shared: ProfileStore {
-		get async {
-			await managedAtomicLazyRef.storeIfNilThenLoad(ProfileStore())
-		}
-	}
+	public static let shared = ProfileStore()
+
+	/// Holds an in-memory copy of the Profile, the source of truth is Keychain.
+	private let profileSubject: AsyncCurrentValueSubject<Profile>
 
-	/// Current Profile
-	let profileStateSubject: AsyncCurrentValueSubject<ProfileState>
+	/// Only mutable since we need to update the description with async, since reading
+	/// device model and name is async.
+	private var deviceInfo: DeviceInfo
 
-	init() async {
-		self.profileStateSubject = await .init(Self.restoreFromSecureStorageIfAble())
+	init() {
+		let metaDeviceInfo = Self._deviceInfo()
+		let stuff = Self._loadSavedElseNewProfile(metaDeviceInfo: metaDeviceInfo)
+		self.deviceInfo = stuff.deviceInfo
+		self.profileSubject = AsyncCurrentValueSubject(stuff.profile)
+
+		if let conflictingOwners = stuff.conflictingOwners {
+			Task {
+				loggerGlobal.notice("ProfileStore:init => emitOwnershipConflict")
+				try await self.emitOwnershipConflict(
+					ownerOfCurrentProfile: conflictingOwners.ownerOfCurrentProfile
```

Comment:
this emits the alert on `overlayWindowClient`, the result of this is that it is seen simultaneously with Keychain Bio Auth prompt... not ideal. Can perhaps "buffer" overlayClient until a new method `ready` has been called, done in Splash after initial unlock bio is done... or something similar.